### PR TITLE
Feature/lead lazy load

### PR DESCRIPTION
**Pull request by thenav56**

### CODE REVIEW

File: apps/lead/tests/test_apis.py [hunk 1]
```diff
@@ -138,10 +173,72 @@ def test_update_assignee(self):
...
+
+        response = self.client.post(url, data)
         self.assert_200(response)
 
+        # Only members are returned when requested is None
+        data = {
+            'projects': [project.pk],
+        }
+        response = self.client.post(url, data)
+        assert response.json(), {
```
Comment (bewakes) [inline apps/lead/tests/test_apis.py]: This looks weird. I think it should have been `response.json() == {....`.

File: apps/lead/tests/test_apis.py [hunk 2]
```diff
@@ -138,10 +173,72 @@ def test_update_assignee(self):
...
+
+        # If value are provided respective data are provided (filtered by permission)
+        data = {
+            'projects': [project.pk],
+            'leadGroups': [lead_group2.pk],
+            'members': [user1.pk, user2.pk, self.user.pk],
+            'organizations': [unhcr.pk]
+        }
+        response = self.client.post(url, data)
+        assert response.json(), {
```
Comment (bewakes) [inline apps/lead/tests/test_apis.py]: Here as well.

File: apps/lead/views.py [hunk 3]
```diff
@@ -67,6 +74,19 @@ def get_queryset(self):
...
+class ProjectLeadGroupViewSet(LeadGroupViewSet):
+    """
+    NOTE: Only to be used by Project's action route [DONOT USE DIRECTLY]
+    """
+    pagination_class = AutocompleteSetPagination
+    serializer_class = SimpleLeadGroupSerializer
+    filter_backends = (filters.SearchFilter,)
+
+    def get_queryset(self):
+        project = Project.objects.get(pk=self.request.query_params['project'])
```
Comment (bewakes) [inline apps/lead/views.py:86]: Maybe use `query_params.get()`?

File: apps/project/tests/test_apis.py [hunk 4]
```diff
@@ -1087,3 +1091,17 @@ def test_project_entries_stats(self):
...
+        lead_group1 = self.create(LeadGroup, project=project)
+        lead_group2 = self.create(LeadGroup, project=project)
+
+        url = f'/api/v1/projects/{project.pk}/lead-groups/'
+
+        self.authenticate()
+        response = self.client.get(url)
+        self.assert_200(response)
+        # Only provide projects leads-group [Pagination is done for larger dataset]
+        assert set([lg['id'] for lg in response.json()['results']]),\
```
Comment (bewakes) [inline apps/project/tests/test_apis.py]: Here also. I  think it should be `assert val1 == val2`, not `assert val1, val2`.

**Review by bewakes — CHANGES_REQUESTED** [review]

File: apps/lead/views.py (hunk 3 above)
Comment (thenav56) [reply apps/lead/views.py:86]: It's defined behavior. It's in the `NOTE:`.
https://github.com/the-deep/server/blob/75d903f58e4947bcb1658a4295781e9f028c11b8/apps/lead/views.py#L77-L87

File: apps/lead/tests/test_apis.py (hunk 1 above)
Comment (thenav56) [reply apps/lead/tests/test_apis.py]: Fixed

File: apps/lead/tests/test_apis.py (hunk 2 above)
Comment (thenav56) [reply apps/lead/tests/test_apis.py]: Fixed

File: apps/project/tests/test_apis.py (hunk 4 above)
Comment (thenav56) [reply apps/project/tests/test_apis.py]: Fixed

**Pull request merged**